nupic/algorithms/CLAClassifier.py

```diff
@@ -558,13 +560,13 @@ def read(cls, proto):
     classifier._patternNZHistory = deque(maxlen=max(classifier.steps) + 1)
     patternNZHistoryProto = proto.patternNZHistory
     learnIteration = classifier._learnIteration - len(patternNZHistoryProto) + 1
-    for i in xrange(len(patternNZHistoryProto)):
+    for i in range(len(patternNZHistoryProto)):
```

Comment:
More appropriate change would have been to leave as-is and use `from six.moves import xrange` (see http://pythonhosted.org/six/#module-six.moves <http://pythonhosted.org/six/#module-six.moves>)

> On Aug 5, 2015, at 2:45 PM, vitaly-krugl <notifications@github.com> wrote:
> 
> In nupic/algorithms/CLAClassifier.py <https://github.com/numenta/nupic/pull/2377#discussion_r36358380>:
> 
> > @@ -558,13 +560,13 @@ def read(cls, proto):
> >      classifier._patternNZHistory = deque(maxlen=max(classifier.steps) + 1)
> >      patternNZHistoryProto = proto.patternNZHistory
> >      learnIteration = classifier._learnIteration - len(patternNZHistoryProto) + 1
> > -    for i in xrange(len(patternNZHistoryProto)):
> > +    for i in range(len(patternNZHistoryProto)):
> This is an example of an automatic change that is inappropriate in most circumstances for python 2.x:
> 
> it adds unnecessary overhead of creating a list on python 2.x
> In the cases where the parameter to xrange was a really large number, this would generate a huge list on python 2.x
> —
> Reply to this email directly or view it on GitHub <https://github.com/numenta/nupic/pull/2377/files#r36358380>.
>